ice_fishing_abm_1/ice_fishing_abm_gp/tests/test_movement_destination_subroutine.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import pytest
+from pytest import fixture
+from ice_fishing_abm_1.ice_fishing_abm_gp.movement_destination_subroutine import ExplorationStrategy, \
+    GPExplorationStrategy
+
+
+@fixture
+def exploration_strategy():
+    return ExplorationStrategy()
+
+
+@fixture
+def gp_exploration_strategy():
+    return GPExplorationStrategy()
+
+
+def test_initialization_exploration_strategy(exploration_strategy):
+    assert exploration_strategy.grid_size == 100
+    assert exploration_strategy.ucb_beta == 0.2
+    assert exploration_strategy.softmax_tau == 0.01
+    assert exploration_strategy.mesh.shape == (10000, 2)
+    assert exploration_strategy.belief_softmax.shape == (100, 100)
+
+
+def test_initialization_gp_exploration_strategy(gp_exploration_strategy):
+    assert gp_exploration_strategy.grid_size == 100
+    assert gp_exploration_strategy.w_social == 0.4
+    assert gp_exploration_strategy.w_success == 0.3
+    assert gp_exploration_strategy.w_failure == 0.3
+    assert gp_exploration_strategy.social_feature_m.shape == (100, 100)
+    assert gp_exploration_strategy.success_feature_std.shape == (100, 100)
+
+
+@pytest.mark.happy
+def test_calculate_gp_feature(gp_exploration_strategy):
+    gp_model = GPExplorationStrategy()
+
+    # empty data case
+    data = np.empty((0, 2))
+    feature_m, feature_std = gp_model._calculate_gp_feature(data, gp_model.social_gpr, gp_model.grid_size)
+
+    assert feature_m.shape == (100, 100)
+    assert feature_std.shape == (100, 100)
+    assert np.all(feature_m == 0)
+    assert np.all(feature_std == 0)
+
+    # not empty data case
+    data = np.array([[10, 10], [20, 20]])
+    feature_m, feature_std = gp_model._calculate_gp_feature(data, gp_model.social_gpr, gp_model.grid_size)
+
+    assert feature_m.shape == (100, 100)
+    assert feature_std.shape == (100, 100)
```

Comment:
It would also be nice to have more test cases with different combinations of social locations. But not necessary